libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -547,97 +517,97 @@ void ESP8266WebServerTemplate<ServerType>::_streamFileCore(const size_t fileSize
-  return _currentArgCount;
+size_t ESP8266WebServerTemplate<ServerType>::args() const {
+    return _currentArgCount;
 }
 
 template <typename ServerType>
 bool ESP8266WebServerTemplate<ServerType>::hasArg(const String& name) const {
-  for (int j = 0; j < _postArgsLen; ++j) {
-    if (_postArgs[j].key == name)
-      return true;
-  }
-  for (int i = 0; i < _currentArgCount; ++i) {
-    if (_currentArgs[i].key == name)
-      return true;
-  }
-  return false;
+    for (size_t j = 0; j < _postArgsLen; ++j) {
+        if (_postArgs[j].key == name)
+            return true;
+    }
+    for (size_t i = 0; i < _currentArgCount; ++i) {
+        if (_currentArgs[i].key == name)
+            return true;
+    }
+    return false;
 }
 
 
 template <typename ServerType>
 const String& ESP8266WebServerTemplate<ServerType>::header(const String& name) const {
-  for (int i = 0; i < _headerKeysCount; ++i) {
-    if (_currentHeaders[i].key.equalsIgnoreCase(name))
-      return _currentHeaders[i].value;
-  }
-  return emptyString;
+    for (size_t i = 0; i < _headerKeysCount; ++i) {
+        if (_currentHeaders[i].key.equalsIgnoreCase(name))
+            return _currentHeaders[i].value;
+    }
+    return emptyString;
 }
 
 template <typename ServerType>
 void ESP8266WebServerTemplate<ServerType>::collectHeaders(const char* headerKeys[], const size_t headerKeysCount) {
-  _headerKeysCount = headerKeysCount + 1;
-  if (_currentHeaders)
-     delete[]_currentHeaders;
-  _currentHeaders = new RequestArgument[_headerKeysCount];
-  _currentHeaders[0].key = FPSTR(AUTHORIZATION_HEADER);
-  for (int i = 1; i < _headerKeysCount; i++){
-    _currentHeaders[i].key = headerKeys[i-1];
-  }
+    _headerKeysCount = headerKeysCount + 1;
+    if (_currentHeaders)
+        delete [] _currentHeaders;
+    _currentHeaders = new RequestArgument[_headerKeysCount];
+    _currentHeaders[0].key = FPSTR(AUTHORIZATION_HEADER);
+    for (size_t i = 1; i < _headerKeysCount; i++) {
+        _currentHeaders[i].key = headerKeys[i-1];
+    }
 }
 
 template <typename ServerType>
-const String& ESP8266WebServerTemplate<ServerType>::header(int i) const {
-  if (i < _headerKeysCount)
-    return _currentHeaders[i].value;
-  return emptyString;
+const String& ESP8266WebServerTemplate<ServerType>::header(size_t i) const {
+    if (i < _headerKeysCount)
+        return _currentHeaders[i].value;
+    return emptyString;
 }
 
 template <typename ServerType>
-const String& ESP8266WebServerTemplate<ServerType>::headerName(int i) const {
-  if (i < _headerKeysCount)
-    return _currentHeaders[i].key;
-  return emptyString;
+const String& ESP8266WebServerTemplate<ServerType>::headerName(size_t i) const {
+    if (i < _headerKeysCount)
+        return _currentHeaders[i].key;
+    return emptyString;
 }
 
 template <typename ServerType>
-int ESP8266WebServerTemplate<ServerType>::headers() const {
-  return _headerKeysCount;
+size_t ESP8266WebServerTemplate<ServerType>::headers() const {
```

Comment:
Signature change

Reply:
return value is not part of the signature, so this isn't actually a signature change.